src/DotNetWorker.Core/Logging/WorkerLogger.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Functions.Worker.Logging.ApplicationInsights;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Functions.Worker.Logging
+{
+    internal class WorkerLogger : ILogger
+    {
+        private readonly string _category;
+        private readonly ISystemLogWriter _systemLogWriter;
+        private readonly IUserLogWriter _userLogWriter;
+        private readonly IUserMetricWriter _userMetricWriter;
+        private readonly IExternalScopeProvider _scopeProvider;
+
+        public WorkerLogger(string category, ISystemLogWriter systemLogWriter, IUserLogWriter userLogWriter, IUserMetricWriter userMetricWriter, IExternalScopeProvider scopeProvider)
+        {
+            _category = category;
+            _systemLogWriter = systemLogWriter;
+            _userLogWriter = userLogWriter;
+            _userMetricWriter = userMetricWriter;
+            _scopeProvider = scopeProvider;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            // The built-in DI wire-up guarantees that scope provider will be set.
+            return _scopeProvider.Push(state);
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel != LogLevel.None;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (WorkerMessage.IsSystemLog)
+            {
+                _systemLogWriter.WriteSystemLog(_scopeProvider, _category, logLevel, eventId, state, exception, formatter);
```

Comment:
Would this log anything ? Because `ISystemLogWriter` implementation registered to DI container is `NullLogWriter` which does nothing.

Reply:
This gets overridden by the call to AddGrpc (it adds the `GrpcFunctionsHostLogWriter` via DI).

System logs are meant to be only for us (in kusto). They have to go back through the host and don't show up in App Insights today. That's why I had to split things out so weirdly -- we needed ways to stop one type of log (user logs) from going through the host, while allowing others (the system logs) to continue flowing.